### PR TITLE
Fix coverage data upload & download

### PR DESCRIPTION
**Pull request by brunns**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -53,27 +53,37 @@ jobs:
...
       - name: Install coverage
         run: python -m pip install --upgrade coverage[toml]
 
+      - name: Merge coverage data
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: coverage-data
+          pattern: coverage-data-*
+          delete-merged: true
+          include-hidden-files: true
```
[the remark this answers is not in the document]
Comment (offbyone) [reply .github/workflows/main.yml:83]: Rrrgh. I added this in the wrong place. It's meant to be on the download.

[the remark this answers is not in the document]
Comment (brunns) [reply .github/workflows/main.yml:83]: Using the separate merge step does leave us with fewer artefacts in the end, but I'm happy to do it this way instead.

[the remark this answers is not in the document]
Comment (offbyone) [reply .github/workflows/main.yml:83]: Nah, as long as it works as-is, good enough for me.

[the remark this answers is not in the document]
Comment (offbyone) [reply .github/workflows/main.yml:83]: Merge it as suits you.

[the remark this answers is not in the document]
Comment (brunns) [reply .github/workflows/main.yml]: Ta.

**Pull request merged**